### PR TITLE
Stability improvements for deformable.

### PR DESCRIPTION
**Pull request by xhan0619**
1) Added Baumgarte stabilization when distance is positive
2) Added split impulse for deformable vs. multibody.
3) Tune parameters for deformable pybullet examples.

### CODE REVIEW

File: src/BulletDynamics/Featherstone/btMultiBody.h [hunk 1]
```diff
@@ -397,6 +402,26 @@ btMultiBody
...
+    void applyDeltaSplitVeeMultiDof(const btScalar *delta_vee, btScalar multiplier)
+    {
+        for (int dof = 0; dof < 6 + getNumDofs(); ++dof)
+        {
+            m_splitV[dof] += delta_vee[dof] * multiplier;
+        }
+    }
+    void addSplitV()
+    {
+        applyDeltaVeeMultiDof(&m_splitV[0], 1);
```
Comment (erwincoumans) [inline src/BulletDynamics/Featherstone/btMultiBody.h:414]: did you want to call applyDeltaSplitVeeMultiDof?
If calling  applyDeltaVeeMultiDof, it may change default behavior of the simulation.

File: src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp [hunk 2]
```diff
@@ -579,6 +579,7 @@ void btMultiBodyDynamicsWorld::integrateMultiBodyTransforms(btScalar timeStep)
 		for (int b = 0; b < m_multiBodies.size(); b++)
 		{
 			btMultiBody* bod = m_multiBodies[b];
+            bod->addSplitV();
```
Comment (erwincoumans) [inline src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp:582]: can this be moved after the sleeping check?

File: src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp [hunk 3]
```diff
@@ -615,6 +616,7 @@ void btMultiBodyDynamicsWorld::integrateMultiBodyTransforms(btScalar timeStep)
 			{
 				bod->clearVelocities();
 			}
+            bod->substractSplitV();
```
Comment (erwincoumans) [inline src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp:619]: same, move within the sleep check, if possible.

File: src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp (hunk 2 above)
Comment (xhan0619) [reply src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp]: Done.

File: src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp (hunk 3 above)
Comment (xhan0619) [reply src/BulletDynamics/Featherstone/btMultiBodyDynamicsWorld.cpp]: Done.

File: src/BulletSoftBody/btSoftBody.cpp [hunk 4]
```diff
@@ -4089,7 +4089,7 @@ void btSoftBody::defaultCollisionHandler(const btCollisionObjectWrapper* pcoWrap
             {
                 const btTransform wtr = pcoWrap->getWorldTransform();
                 const btScalar timemargin = 0;
-                const btScalar basemargin = getCollisionShape()->getMargin();
+                const btScalar basemargin = 3.0*getCollisionShape()->getMargin();
                 btVector3 mins;
```
Comment (erwincoumans) [inline src/BulletSoftBody/btSoftBody.cpp:4093]: there are more magic constants, like 2, 3 and 4. Why are those needed? If they are important, it should be exposed to the user?

Comment (xhan0619) [reply src/BulletSoftBody/btSoftBody.cpp]: Those are more experimental. I have removed them from this PR.
